test/test_sklearn_compat.py

```diff
@@ -79,6 +80,9 @@ def test_sdml(self):
   def test_rca(self):
     check_estimator(Stable_RCA_Supervised)
 
+  def test_scml(self):
+    check_estimator(SCML_Supervised)
+
```

Comment:
I added this test but it is currently failing. The dataset used has very close values for the number of samples and of features, this leads to a large number of basis, which is currently allowed but yields multiple bases that are the same. I am not yet seeing why, but this yields the same values for the weights instead of just keeping one, which should be the behaviour under _l1_ regularization. If anyone has any insight on this it would be of great help.

Reply:
Why is the test failing exactly?

Reply:
On the Cholesky decomposition for the computation of `L`, it fails as the yielded matrix is not PSD. It caught my attention that all the bases were the same and the weights too. Very weird behaviour.

Reply:
- *All* bases are the same, or some of them? If it is all of them it is very weird indeed, you should look into it to find out what happens
- Even if all bases are the same, as long as the weights are positive the final matrix should be PSD? Did you look into it more deeply to understand what happens? It is probably some numerical issue, e.g. some eigenvalue super close to zero but negative? This makes sense, and in fact in the case where `n_basis >= n_features` you should probably fall back on `components_from_metric` for a more robust behavior

Reply:
I tried to change the `n_basis` and I could achieve different escenarios with repeating basis, and with the same shared weigths on each set of repeated basis. I have to take a deeper look to see what is really going on. 

Yes, that is exactly what happens. If I compute the `np.linalg.eig` of `M` it yields some `e-17` negative values. That would definitely fix this really quickly!

Reply:
As the bases are exactly the same and the weights are all initialized on zero the effect of each iteration on the weights is shared.

I think this could be either solved by initializing the weights by random values or running an `np.unique` over the basis set. I think the first option would be better as the second would be very costly, but I don't know what the effect of changing the initialization would be.

Reply:
When initializing with uniform weights it is indeed normal to get the same weight for bases that are the same. For *identical bases*, any allocation of a fixed amount of weight magnitude has the same l1 norm (w1=2 and w2=2, or w1=1 and w2=3, or w1=0 and w2=4, etc). The l1 norm eliminates linearly related features only when they are not exactly equal.

Of course it is very surprising that if you initialize to random weights you also get equal weights at the end !?